pallets/inflation-manager/src/lib.rs

```diff
@@ -268,5 +267,79 @@ pub mod pallet {
 
 			InflationParametersT { inflation_rate, disinflation_rate }
 		}
+
+		pub fn fund_difference_balances() {
+			let account = T::PotId::get().into_account_truncating();
+			let now_total_issuance = T::Currency::total_issuance();
+			let desired_issuance = T::DefaultTotalIssuanceNum::get();
+			if now_total_issuance < desired_issuance {
+				let amount = desired_issuance.saturating_sub(now_total_issuance);
+				T::Currency::deposit_creating(&account, amount);
+				log::info!(
+					"Total issuance was increased from {:?} to {:?}, by {:?} tokens.",
+					now_total_issuance,
+					desired_issuance,
+					amount
+				);
+			}
+		}
+
+		pub fn initialize_inflation() -> Weight {
+			let current_block = frame_system::Pallet::<T>::block_number();
+			let mut weight_reads = 1;
+			let mut weight_writes = 0;
+
+			let inflation_configuration = T::DefaultInflationConfiguration::get();
+			// install inflation config
+			InflationConfiguration::<T>::put(inflation_configuration.clone());
+			weight_writes += 1;
+
+			// set current year to 1
+			CurrentYear::<T>::put(1);
+			weight_writes += 1;
+
+			// calc inflation for first year
+			let inflation_parameters =
+				Pallet::<T>::update_inflation_parameters(&inflation_configuration);
+			weight_reads += 1;
+
+			// install inflation parameters for first year
+			InflationParameters::<T>::put(inflation_parameters.clone());
+			weight_writes += 1;
+
+			// set the flag to calculate inflation parameters after a year(in blocks)
+			let racalculation_target_block = current_block + T::BlockNumber::from(BLOCKS_PER_YEAR);
+
+			// Update recalculation flag
+			DoRecalculationAt::<T>::put(racalculation_target_block);
+			weight_writes += 1;
+
+			let block_rewards = Pallet::<T>::rewards_per_block(&inflation_parameters);
+			weight_reads += 1;
+
+			BlockRewards::<T>::put(block_rewards);
+			weight_writes += 1;
+
+			T::DbWeight::get().reads_writes(weight_reads, weight_writes)
+		}
+
+		/// Sets DoRecalculationAt to the given block number where year 1 will kick off
+		pub fn initialize_delayed_inflation(do_recalculation_at: T::BlockNumber) -> Weight {
```

Comment:
Could you explain this logic more? Could you tell me whether my thinking is correct or not
1. In the initialized delay inflation, we will start from the current year 0, the DoRecalculationAt (200), and set the InflationConfiguration by the default inflation configuration
2. When we go to the on_finalized at 200 blocks, we'll update_inflation_parameters based on the inflation configuration and set up the current year as 1.

In the current year 1, there might be two different situations.
From initialized inflation: year 1 is the default inflation configuration.
From initialized delayed inflation: year 1 is the updated inflation configuration based on the default inflation configuration

That's what we want, right

Reply:
Got it! Thanks a lot!